lib/pbench/test/functional/server/test_put.py

```diff
@@ -35,44 +39,61 @@ def test_upload_all(self, server_client: PbenchServerClient, login_user):
         datasets = server_client.get_list(
             metadata=["dataset.access", "server.tarball-path", "server.status"]
         )
-        found = sorted(d.name for d in datasets)
-        expected = sorted(self.tarballs.keys())
-        assert found == expected
-        for dataset in datasets:
-            t = self.tarballs[dataset.name]
-            assert dataset.name in dataset.metadata["server.tarball-path"]
-            assert dataset.metadata["server.status"]["upload"] == "ok"
-            assert t.access == dataset.metadata["dataset.access"]
+        found = {d.name for d in datasets}
+        expected = frozenset(self.tarballs.keys())
+        assert expected.issubset(found), f"expected {expected!r}, found {found!r}"
+        try:
+            for dataset in datasets:
+                if dataset.name not in expected:
+                    continue
+                t = self.tarballs[dataset.name]
+                assert dataset.name in dataset.metadata["server.tarball-path"]
+                assert dataset.metadata["server.status"]["upload"] == "ok"
+                assert t.access == dataset.metadata["dataset.access"]
+        except HTTPError as exc:
```

Comment:
How is this possible? We might get an error on the `server_client.get_list()`, but that's not within a `try` (which was intentional, since we just want the test to fail if that happens anyway).

Reply:
Yeah, I thought about that, too. What it does is add inline verbiage of questionable value to change the way an error is reported, rather than simply letting the exception fly. While I would be more inclined to just use the exception to fail, I was also slightly less inclined to complain than here where the `try` block serves no purpose at all.

Reply:
Generators ...

Reply:
Ha; yeah, and I'm the one who turned `get_list` into a generator, too. 😬